lib/slots.ts

```diff
@@ -12,17 +12,69 @@ const getMinutesFromMidnight = (date) => {
   return date.hour() * 60 + date.minute();
 };
 
+interface SlotAvailability {
+  available: boolean,
+  offset: number,
+}
+
+const isSlotAvailable = ({
+  slotStartTime,
+  slotEndTime,
+  busyTimes
+}): SlotAvailability => {
+  let slotAvailability: SlotAvailability = {
+    available: true,
+    offset: 0,
+  }
+
+  for (let i = 0; i < busyTimes.length; i += 1) {
+    const busyTimeStart = dayjs(busyTimes[i].start);
+    const busyTimeEnd = dayjs(busyTimes[i].end);
+
+    // Check if start times are the same
+    if (slotStartTime.format('HH:mm') == busyTimeStart.format('HH:mm')) {
+      slotAvailability.available = false;
+      slotAvailability.offset = busyTimeEnd.diff(slotStartTime, 'minute');
+    }
+
+    // Check if time is between start and end times
+    else if (slotStartTime.isBetween(busyTimeStart, busyTimeEnd)) {
+      slotAvailability.available = false;
+      slotAvailability.offset = busyTimeEnd.diff(slotStartTime, 'minute');
+    }
+
+    // Check if slot end time is between start and end time
+    else if (slotEndTime.isBetween(busyTimeStart, busyTimeEnd)) {
+      slotAvailability.available = false;
+      slotAvailability.offset = busyTimeEnd.diff(slotStartTime, 'minute');
+    }
+
+    // Check if startTime is between slot 
+    else if (busyTimeStart.isBetween(slotStartTime, slotEndTime)) {
+      slotAvailability.available = false;
+      slotAvailability.offset = busyTimeEnd.diff(slotStartTime, 'minute');
+    }
+
+    if (!slotAvailability.available) {
+      if(slotAvailability.offset === 0) slotAvailability.offset = 5;
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689309)
```suggestion
      if(slotAvailability.offset === 0) {slotAvailability.offset = 5;}
```